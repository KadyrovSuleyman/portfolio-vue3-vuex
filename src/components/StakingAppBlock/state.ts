import { Store } from 'vuex';

export interface StateT {
  isWalletApproved: boolean,
  isStaked: boolean,
  viewContract: () => void,
}

export const adaptState = (store: Store<any>): StateT => ({
  isWalletApproved: store.state.wallet.isApproved,
  isStaked: store.state.stake.isStaked,
  viewContract: () => store.commit('stake/mockCountdown', 3000),
});
