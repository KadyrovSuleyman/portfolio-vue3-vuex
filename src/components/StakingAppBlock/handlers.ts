import { ComputedRef } from 'vue';
import { StateT } from './state';

const createViewContractClickHandler = (state: ComputedRef<StateT>) => () => {
  state.value.viewContract();
};

export default createViewContractClickHandler;
