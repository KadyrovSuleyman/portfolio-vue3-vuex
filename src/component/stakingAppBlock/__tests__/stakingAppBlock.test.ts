import { mount, VueWrapper } from '@vue/test-utils';
import { createStore, Store } from 'vuex';
import StakingAppBlock from '../stakingAppBlock.vue';

jest.mock('../adapter');
jest.mock('../actionsBlock/adapter.ts');
jest.mock('../calculatorBlock/adapter.ts');
jest.mock('../stakeInfoBlock/infoContainer/adapter.ts');
jest.mock('../stakeInfoBlock/adapter.ts');
jest.mock('../tariffsBlock/adapter.ts');

let store: Store<any>;
beforeEach(() => {
  store = createStore<any>({
    state: {
      isStaked: false,
    },
    mutations: {
      change: (state, obj: { [name: string]: boolean | string }) => {
        Object.keys(obj).forEach((index) => {
          state[index] = obj[index];
        });
      },
    },
  });
});

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

// =================================
it('stakingAppBlock renders', () => {
  wrapper = mount(StakingAppBlock, {
    global: { plugins: [store] },
  });
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});

it('watchs props changes', async () => {
  wrapper = mount(StakingAppBlock, {
    global: { plugins: [store] },
  });
  expect(wrapper.find('.stakingAppBlock').classes()).toEqual(['stakingAppBlock']);

  await wrapper.setProps({
    mods: {
      selected: true,
    },
  });
  expect(wrapper.find('.stakingAppBlock').classes()).toEqual([
    'stakingAppBlock',
    'stakingAppBlock__selected',
  ]);

  await wrapper.setProps({
    mods: {
      selected: false,
    },
  });
  expect(wrapper.find('.stakingAppBlock').classes()).toEqual(['stakingAppBlock']);
});

it('watching outer store', async () => {
  wrapper = mount(StakingAppBlock, {
    global: { plugins: [store] },
  });
  expect(wrapper.element.outerHTML).toMatchSnapshot();

  store.commit('change', {
    isWalletApproved: true,
  });
  await wrapper.vm.$nextTick();
  expect(wrapper.element.outerHTML).toMatchSnapshot();

  store.commit('change', {
    isWalletApproved: false,
  });
  await wrapper.vm.$nextTick();
  expect(wrapper.element.outerHTML).toMatchSnapshot();

  store.commit('change', {
    isStaked: true,
  });
  await wrapper.vm.$nextTick();
  expect(wrapper.element.outerHTML).toMatchSnapshot();

  store.commit('change', {
    isStaked: false,
  });
  await wrapper.vm.$nextTick();
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});
