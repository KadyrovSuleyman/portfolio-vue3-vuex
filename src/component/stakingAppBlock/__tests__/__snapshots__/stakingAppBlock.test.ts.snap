// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stakingAppBlock renders 1`] = `
<div class="stakingAppBlock"><span class="stakingAppBlock-header"> Staking App </span>
  <!--v-if-->
  <div class="stakingAppBlock-tariffsBlock"></div>
  <div class="stakingAppBlock-calculatorBlock">
    <div class="calculatorBlock-infoBlock">
      <div class="infoBlock-div infoBlock-div__icon_warning"></div><span class="infoBlock-span"> To perform actions on the page, connect your wallet </span>
    </div>
  </div>
  <div class="stakingAppBlock-actionsBlock">
    <!--v-if--><button class="actionsBlock-mainButton">
      <!--v-if--><span class="mainButton-span">Connect wallet</span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if--><button class="stakingAppBlock-viewContractButton"><span class="viewContractButton-span"> View contract </span></button>
</div>
`;

exports[`watching outer store 1`] = `
<div class="stakingAppBlock"><span class="stakingAppBlock-header"> Staking App </span>
  <!--v-if-->
  <div class="stakingAppBlock-tariffsBlock"></div>
  <div class="stakingAppBlock-calculatorBlock">
    <div class="calculatorBlock-infoBlock">
      <div class="infoBlock-div infoBlock-div__icon_warning"></div><span class="infoBlock-span"> To perform actions on the page, connect your wallet </span>
    </div>
  </div>
  <div class="stakingAppBlock-actionsBlock">
    <!--v-if--><button class="actionsBlock-mainButton">
      <!--v-if--><span class="mainButton-span">Connect wallet</span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if--><button class="stakingAppBlock-viewContractButton"><span class="viewContractButton-span"> View contract </span></button>
</div>
`;

exports[`watching outer store 2`] = `
<div class="stakingAppBlock"><span class="stakingAppBlock-header"> Staking App </span>
  <div class="stakingAppBlock-walletApprovedBlock"><span class="walletApprovedBlock-span"> Wallet approved </span>
    <div class="walletApprovedBlock-div walletApprovedBlock-div__icon_check"></div>
  </div>
  <div class="stakingAppBlock-tariffsBlock"></div>
  <div class="stakingAppBlock-calculatorBlock">
    <div class="calculatorBlock-infoBlock">
      <div class="infoBlock-div infoBlock-div__icon_warning"></div><span class="infoBlock-span"> To perform actions on the page, connect your wallet </span>
    </div>
  </div>
  <div class="stakingAppBlock-actionsBlock">
    <!--v-if--><button class="actionsBlock-mainButton">
      <!--v-if--><span class="mainButton-span">Connect wallet</span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if--><button class="stakingAppBlock-viewContractButton"><span class="viewContractButton-span"> View contract </span></button>
</div>
`;

exports[`watching outer store 3`] = `
<div class="stakingAppBlock"><span class="stakingAppBlock-header"> Staking App </span>
  <!--v-if-->
  <div class="stakingAppBlock-tariffsBlock"></div>
  <div class="stakingAppBlock-calculatorBlock">
    <div class="calculatorBlock-infoBlock">
      <div class="infoBlock-div infoBlock-div__icon_warning"></div><span class="infoBlock-span"> To perform actions on the page, connect your wallet </span>
    </div>
  </div>
  <div class="stakingAppBlock-actionsBlock">
    <!--v-if--><button class="actionsBlock-mainButton">
      <!--v-if--><span class="mainButton-span">Connect wallet</span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if--><button class="stakingAppBlock-viewContractButton"><span class="viewContractButton-span"> View contract </span></button>
</div>
`;

exports[`watching outer store 4`] = `
<div class="stakingAppBlock"><span class="stakingAppBlock-header"> Staking App </span>
  <!--v-if-->
  <div class="stakingAppBlock-stakeInfoBlock">
    <div class="stakeInfoBlock-div"><span class="stakeInfoBlock-incomeSpan"></span><span class="stakeInfoBlock-promtSpan"> TKN earned </span></div>
    <div class="stakeInfoBlock-infoContainer">
      <div class="infoContainer-infoItem"><span class="infoItem-header">isStaked</span><span class="infoItem-value"> true</span></div>
      <div class="infoContainer-infoItem"><span class="infoItem-header">isWalletApproved</span><span class="infoItem-value"> false</span></div>
    </div>
  </div>
  <!--v-if-->
  <div class="stakingAppBlock-actionsBlock"><button class="actionsBlock-timeButton"><span class="timeButton-span"></span></button><button class="actionsBlock-mainButton">
      <!--v-if--><span class="mainButton-span">Connect wallet</span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="stakingAppBlock-bgImg"></div><button class="stakingAppBlock-viewContractButton"><span class="viewContractButton-span"> View contract </span></button>
</div>
`;

exports[`watching outer store 5`] = `
<div class="stakingAppBlock"><span class="stakingAppBlock-header"> Staking App </span>
  <!--v-if-->
  <div class="stakingAppBlock-tariffsBlock"></div>
  <div class="stakingAppBlock-calculatorBlock">
    <div class="calculatorBlock-infoBlock">
      <div class="infoBlock-div infoBlock-div__icon_warning"></div><span class="infoBlock-span"> To perform actions on the page, connect your wallet </span>
    </div>
  </div>
  <div class="stakingAppBlock-actionsBlock">
    <!--v-if--><button class="actionsBlock-mainButton">
      <!--v-if--><span class="mainButton-span">Connect wallet</span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if--><button class="stakingAppBlock-viewContractButton"><span class="viewContractButton-span"> View contract </span></button>
</div>
`;
